### PR TITLE
Add publish command

### PR DESCRIPTION
**Pull request by dirtybit**

### CODE REVIEW

**Comment by dirtybit** [comment]
[zem publish](https://trello.com/c/iICEtKB7/61-zem-publish)

File: src/commands/publish/zeplin-api.js [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+            this.setToken(token);
+
+            return await this.getUserProfile();
+        } catch (err) {
+            throw err;
+        }
+    }
+
+    loginUser(handle, password) {
```
Comment (ceylanismail) [inline src/commands/publish/zeplin-api.js]: Note; to talk for token restriction.

File: src/commands/publish/index.js [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                description,
+                projectTypes: projectTypes.join(","),
+                packageBuffer
+            });
+        } else {
+            await apiClient.createExtensionVersion(extension._id, version, packageBuffer);
+        }
+
+        console.log(chalk.blue(`${packageName}@${version} is submitted!\n`));
+        console.log("You'll be notified once it's published on extensions.zeplin.io.");
```
Comment (mertkahyaoglu) [inline src/commands/publish/index.js]: Adding https prefix to the link makes it clickable for some terminals. Also we can chalk it up with orange? 😄

File: src/commands/publish/index.js [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+                name,
+                description,
+                projectTypes: projectTypes.join(","),
+                packageBuffer
+            });
+        } else {
+            await apiClient.createExtensionVersion(extension._id, version, packageBuffer);
+        }
+
+        console.log(chalk.blue(`${packageName}@${version} is submitted!\n`));
```
Comment (mertkahyaoglu) [inline src/commands/publish/index.js]: Should we mention about whether creating for the first time or releasing a version?

Comment (dirtybit) [reply src/commands/publish/index.js]: Good idea 👍 

@berkcebi would you like to work on the copies we add with publish command? 🙄

File: src/commands/publish/index.js (hunk 2 above)
Comment (dirtybit) [reply src/commands/publish/index.js]: @berkcebi

File: README.md [hunk 4]
```diff
@@ -70,6 +70,19 @@ Options:
 
 Cleans build directory.
 
+
+#### `npm run publish`
```
Comment (dirtybit) [inline README.md]: @berkcebi ping

File: src/commands/publish/index.js (hunk 2 above)
Comment (berkcebi) [reply src/commands/publish/index.js]: Adding `https://` definitely makes sense, I'm not sure about the color though as a lot of people use different terminal background colors. 😅 Can we make it bold, or underline it though?

File: src/commands/publish/index.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+    if (!fs.existsSync(manifestPath)) {
+        throw new Error("Could not find manifest.json file!");
+    }
+
+    const manifest = JSON.parse(fs.readFileSync(manifestPath));
+    const { valid, validationErrors } = manifestValidator(manifest);
+
+    if (!valid) {
+        throw new Error(`Invalid manifest.json file:\n${validationErrors}`);
```
Comment (berkcebi) [inline src/commands/publish/index.js]: > Validating **manifest.json** failed:

File: src/commands/publish/index.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+        return paths.resolveBuildPath(relativePath);
+    }
+};
+
+function parseManifest() {
+    const manifestPath = pathResolver.resolve("./manifest.json");
+
+    if (!fs.existsSync(manifestPath)) {
+        throw new Error("Could not find manifest.json file!");
```
Comment (berkcebi) [inline src/commands/publish/index.js]: > Locating **manifest.json** failed.

Both for this and the error below, could we suggest something like re-building? What could typically go wrong for these errors?

File: src/commands/publish/index.js (hunk 3 above)
Comment (berkcebi) [inline src/commands/publish/index.js]: > **Extension Name** (0.1.0) is now submitted. 🏄‍♂️

For versions:

> Version **0.1.0** of **Extension Name** extension is now submitted. 🏄‍♂️

File: src/commands/publish/index.js [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+                packageBuffer
+            });
+        } else {
+            await apiClient.createExtensionVersion(extension._id, version, packageBuffer);
+        }
+
+        console.log(chalk.blue(`${packageName}@${version} is submitted!\n`));
+        console.log("You'll be notified once it's published on extensions.zeplin.io.");
+    } catch (error) {
+        console.log(chalk.red("Couldn't publish the extension:"));
```
Comment (berkcebi) [inline src/commands/publish/index.js]: > Publishing extension failed:

File: src/index.js [hunk 8]
```diff
@@ -70,6 +70,16 @@ program
         exec(require("./config/webpack.dev"), fnName, defaultOptions, options.build);
     });
 
+program
+    .command("publish")
+    .description("Publish extension.")
```
Comment (berkcebi) [inline src/index.js]: > Publish extension, sending it for review to be listed on https://extensions.zeplin.io.

File: src/index.js [hunk 9]
```diff
@@ -70,6 +70,16 @@ program
...
+    .option("--path <build-path>", `Path to the extension build to be published.`)
```
Comment (berkcebi) [inline src/index.js]: > Path for the extension build to be published

No `.`. 😬

File: README.md [hunk 10]
```diff
@@ -70,6 +70,19 @@ Options:
...
+
+Publish extension build to the zem's registry.
```
Comment (berkcebi) [inline README.md]: Publish extension, sending it for review to be listed on https://extensions.zeplin.io.

File: README.md [hunk 11]
```diff
@@ -70,6 +70,19 @@ Options:
...
+
+```
+Usage: npm run publish -- [options]
+
+Options:
+
+  --path <build-path>           Path to look for extension build files. It uses the path used by build command by default.
```
Comment (berkcebi) [inline README.md]: > Path for the extension build to publish (default: Path used by the build command)

**Pull request merged**